### PR TITLE
[FRONT-BACK] Improve Translations and fix tasks issues

### LINKED ISSUE #128: [BACKEND] Leveling & XP Progression System
### Description 
- Implement the logic that calculates experience gain and manages user "Level Ups" after matches.

### Tasks

- [x] Define an XP threshold scale (e.g., Level 1 to 2 = 500 XP).

- [x] Create a service to calculate XP rewards based on match outcomes (Victory bonus vs. Participation XP).

- [ ] Implement the logic to update the User level in the database when an XP threshold is crossed.

- [ ] Return a "Level Up" flag in the API response to notify the frontend when a player gains a level.

### PR DESCRIPTION
**Pull request by nasdhn**
## Summary

This PR bundles several quality-of-life and progression improvements: profile navigation from the Friends page, a reworked exponential levelsystem with updated XP rewards, a global rank badge on every profile, and a complete pass on all three language translations including a fix to the Axios interceptor so backend success messages are now translated
automatically.

---

## Technical Implementation

**Profile navigation:** The Friends page now supports navigating to any user's profile directly from search results, incoming requests, and outgoing requests, making the `/profile/:username` route accessible from everywhere in the app.

**Exponential level system:** XP rewards were updated backend-side (win → 100, draw → 50, loss → 25) and a new pure utility module `levelUtils.ts` was introduced frontend-side. It computes level, current XP within a level, XP needed for the next level, and fill percentage using the formula `floor(100 × N^1.5)`. Early levels progress quickly; higher levels require significantly more XP.

**Global rank:** A new `GET /users/:id/rank` endpoint counts how many users have strictly more XP than the target user and returns `rank = count + 1`. The rank badge is displayed directly on every profile, giving users a clear sense of their standing even if they don't appear in the top-10 leaderboard.

**Translations:** All three language files (EN, FR, ES) were audited and completed. The Axios interceptor was extended to also translate backend success messages (previously it only handled errors), using the same `backend.*` key pattern. This covers messages such as `FRIEND_REMOVED_SUCCESS`, `AUTH_LOGOUT_SUCCESS`, and `AUTH_2FA_ENABLED_SUCCESS`.

---

## Modified Files

| File | Description |
|------|-------------|
| `matches.service.ts` | XP rewards updated: win +100, draw +50, loss +25 |
| `users.service.ts` | Added `getUserRank()` — counts users with strictly higher XP |
| `users.controller.ts` | Added `GET :id/rank` endpoint, placed before `GET :id` |
| `userService.ts` | Added `getUserRank()` frontend service function |
| `src/lib/levelUtils.ts` | New file — exponential level formula (`100 × N^1.5`), `getLevel()`, `getLevelProgress()` |
| `Level.tsx` | Refactored to use `levelUtils`, displays current XP / XP needed |
| `Profile.tsx` | Added rank badge fetched from the new endpoint |
| `Friends.tsx` | Search results, incoming and outgoing requests now navigate to `/profile/:username` |
| `api.ts` | Interceptor success handler extended to translate `data.message` via `backend.*` keys |
| `fr.json` / `en.json` / `es.json` | Full translation audit — missing keys added across all three files |

---

### Environment Setup

```bash
make nuke
make setup
make up
```

Create at least 3 user accounts and play a few matches to generate
real data before testing.

---

### 1. Backend — Validation via Swagger

Open the Swagger UI at:
```
http://localhost:1024/api/api-docs
```

Log in through the frontend first so your `access_token` cookie is set,
then Swagger will automatically use it for all requests.

**Endpoints to test:**

- `GET /users/{id}/rank` with a valid user ID → expect `200` with
  `{ "rank": X, "xp": Y }`
- `GET /users/{id}/rank` with an unknown ID (e.g. `999`) → expect
  `404` with `ERR_USER_NOT_FOUND`

---

### 2. Profile Navigation from Friends

Go to `http://localhost:1024/friends`.

- **Search** for a username → click their avatar or name → lands on
  their `/profile/:username` page in visitor mode.
- **Incoming requests** → click the sender → same result.
- **Outgoing requests** → click the receiver → same result.
- **Friends list** → click a friend → same result.

Verify on the visited profile that edit buttons, avatar upload, and 2FA
section are **hidden** (visitor mode).

---

### 3. Level System & XP

Play a match:

- **Win** → +100 XP. A first win should immediately push a new account
  to level 1 (requires 100 XP).
- **Draw** → +50 XP.
- **Loss** → +25 XP.

On the profile, verify:

- The level label updates correctly after each match.
- The XP bar shows `currentXp / neededXp` (not `/100` anymore).
- Progression slows down as level increases:

````
Level 0 → 1 :  100 XP  (1 win)
Level 1 → 2 :  183 XP  (~2 wins)
Level 2 → 3 :  247 XP  (~3 wins)
````

---

### 4. Global Rank Badge

Go to `http://localhost:1024/profile` (your own profile) or visit
another user's profile via `/profile/:username`.

Verify the amber `Rank #X` badge appears just below the username.

If you gain XP and overtake another player, refresh the profile and
confirm the rank updates.

---

### 5. Translations

Switch between EN, FR and ES using the language switcher.

Navigate through the full app and verify:

- All UI labels, buttons, and error messages are translated.
- Play a match → win/loss/draw toasts appear in the selected language.
- Add/remove a friend → success toast appears translated.
- Log out → success toast appears translated.
- Trigger a backend error (e.g. send a friend request to yourself)
  → error message appears translated.

---

## Expected Behavior

- Every profile shows a `Rank #X` badge regardless of leaderboard
  position.
- First win instantly reaches level 1; progression slows exponentially
  from there.
- Clicking any user anywhere in the Friends page navigates to their
  public profile.
- All user-facing text is translated in EN, FR and ES, including
  backend success and error messages.

---

### Done
- [x] Profile navigation from Friends (search, incoming, outgoing, list)
- [x] XP rewards updated (win 100 / draw 50 / loss 25)
- [x] Exponential level formula (`levelUtils.ts`)
- [x] `Level.tsx` updated with dynamic XP-to-next-level display
- [x] `GET /users/:id/rank` backend endpoint
- [x] Rank badge on all profiles
- [x] Axios interceptor extended to translate success messages
- [x] Full translation audit — EN / FR / ES completed

### Notes
Closes #292 
Closes #128

### CODE REVIEW

**Review by Nico-Ry — APPROVED** [review]
Tested everything described in the PR and overall everything works really well 🚀
This is a super nice improvement to the progression system and profile experience overall 👏

## ✅ What I tested

### Friends → Profile navigation

Tested navigation from:

* search results
* incoming requests
* outgoing requests
* friends list

All correctly redirect to `/profile/:username` and visitor mode behaves properly (no edit buttons / no 2FA controls visible).

### Rank system

Tested the new `GET /users/:id/rank` endpoint through Swagger and frontend integration.

* valid users return correct rank/xp
* invalid IDs correctly return `404`
* rank badge updates correctly after gaining XP

Very nice addition honestly, profiles feel much more alive now 🏆

### XP / Level system

Played multiple matches and verified:

* win = +100 XP
* draw = +50 XP
* loss = +25 XP

Also checked:

* level progression updates correctly
* XP bar now uses dynamic values
* exponential scaling behaves as expected

The new progression curve feels MUCH better than before 📈

### Translations

Tested EN / FR / ES:

* frontend labels
* backend errors
* backend success messages
* friend actions
* logout
* gameplay notifications

Everything translated correctly from what I tested 🌍

---

## ⚠️ Small issue found (non-blocking)

The only issue I noticed is with seeded dummy users.

Some dummy accounts that already have seeded matches:

* do not have achievements correctly unlocked
* have XP values that do not fully match the new XP system introduced in this PR

Real gameplay works correctly, so this is not blocking at all for merging 👍
I think we probably just need to update the seed logic/data so dummy users are generated consistently with the new progression system.

Great work overall, approving 🚀
